public/js/common.js

```diff
@@ -68,3 +68,113 @@ function failSnackbar(msg) {
 $(document).on('click', '#closeSnack', function() {
     $(this).parent().fadeOut();
 });
+
+
+/*
+These errors are only for UI use
+Error codes and their corresponding message. Error codes are under different
+categories:
+1000 -> system
+2000 -> user
+3000 -> question
+4000 -> class
+5000 -> analytics
+6000 -> import/export
+7000 -> settings
+*/
+
+const errors = Object.freeze({
+    //1000 system
+    1000: 'Invalid Request',
+    1001: 'Missing Requirement',
+    1002: 'Permission Denied',
+    1003: 'Confirm password doesn\'t match',
+    1004: 'Database connection failed',
+    1005: 'Database Error',
+    1006: 'Invalid username or password',
+    1007: 'A system error has occured',
+    1008: 'A system error has occured',
+    1009: 'A system error has occured',
+    1010: 'A system error has occured',
+    1011: 'A system error has occured',
+    1012: 'A system error has occured',
+    1013: 'A system error has occured',
+
+    //2000 user
+    2000: 'Invalid username or password',
+    2001: 'Failed to get student',
```

Comment:
if this was triggered by invalid Id then its a user issue, but if it was triggered due to our backend then its most likely a db issue

Reply:
What would you like to change the error to?